### PR TITLE
Add S6439: React components should not render non-boolean condition values

### LINKED ISSUE #3205: Add rule S6439: React components should not render non-boolean condition values
[RSPEC PR](https://github.com/SonarSource/rspec/pull/1091)

We should try first `react/jsx-no-leaked-render` as implementation but there is a big chance it will be noisy as it's not relying on type information.

### PR DESCRIPTION
**Pull request by victor-diez-sonarsource**
Fixes #3205

### CODE REVIEW

**Review by saberduck — CHANGES_REQUESTED** [review]

File: eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { Rule, SourceCode } from 'eslint';
+import * as estree from 'estree';
+import { getTypeFromTreeNode, isBigIntType, isNumberType, isStringType } from '../utils';
+
+export const rule: Rule.RuleModule = {
+  meta: {
+    hasSuggestions: true,
+    messages: {
+      noPotentialLeakedRender:
+        'Potential leaked value that might cause unintentionally rendered values or rendering crashes',
```
Comment (saberduck) [inline eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts:31]: Issue message should be actionable, from this message it is not clear to the user how to fix the problem. I would rephrase it to something like "Convert value to a boolean type to avoid rendering issues"

File: eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      suggestCoercion: 'Coerce the conditional to a boolean',
```
Comment (saberduck) [inline eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts:32]: I would replace "coerce" to "convert", it's a more common and easier-to-understand word.

**Review by saberduck — CHANGES_REQUESTED** [review]

File: eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts (hunk 1 above)
Comment (victor-diez-sonarsource) [reply eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts:31]: just kept the message from original eslint plugin implementation. I'll modify that

File: eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts (hunk 2 above)
Comment (victor-diez-sonarsource) [reply eslint-bridge/src/rules/sonar-jsx-no-leaked-render.ts:32]: same as the message, I kept the suggestion from the original. I'll change that now

**Review by saberduck — APPROVED** [review]
